pkg/config/validation/validation.go

```diff
@@ -1306,15 +1306,38 @@ func validateTrafficPolicy(policy *networking.TrafficPolicy) Validation {
 		return Validation{}
 	}
 	if policy.OutlierDetection == nil && policy.ConnectionPool == nil &&
-		policy.LoadBalancer == nil && policy.Tls == nil && policy.PortLevelSettings == nil {
+		policy.LoadBalancer == nil && policy.Tls == nil && policy.PortLevelSettings == nil && policy.Tunnel == nil {
 		return WrapError(fmt.Errorf("traffic policy must have at least one field"))
 	}
 
 	return appendValidation(validateOutlierDetection(policy.OutlierDetection),
 		validateConnectionPool(policy.ConnectionPool),
 		validateLoadBalancer(policy.LoadBalancer),
 		validateTLS(policy.Tls),
-		validatePortTrafficPolicies(policy.PortLevelSettings))
+		validatePortTrafficPolicies(policy.PortLevelSettings),
+		validateTunnelSettings(policy.Tunnel))
+}
+
+func validateTunnelSettings(tunnel *networking.TrafficPolicy_TunnelSettings) (errs error) {
+	if tunnel == nil {
+		return
+	}
+	if tunnel.Protocol == "" {
+		errs = appendErrors(errs, fmt.Errorf("tunnel protocol must be specified"))
+	}
+	if tunnel.Protocol != "connect" && tunnel.Protocol != "post" {
+		errs = appendErrors(errs, fmt.Errorf("tunnel protocol must be \"connect\" or \"post\""))
+	}
+	if tunnel.TargetHost == "" {
+		errs = appendErrors(errs, fmt.Errorf("tunnel target host must be specified"))
+	}
+	if strings.Contains(tunnel.TargetHost, "*") {
```

Comment:
can we make this a bit more robust? this would allow something like `.....~~\/%^&` which is certainly not valid. We have some helpers like `ValidateFQDN` but I don't think we have one for "FQDN or IP" but could add one